Tests fail on a system OS set up in languages other than English

**Describe the bug**

When trying to execute `make test` on an OS set up in Spanish language, all the git error messages show up in Spanish, making the tests checking for those messages in English fail.

**To Reproduce**

- Set up your OS in a language different than English
- Execute `make test`

**Expected behavior**

We should be able to execute the test suite independently of the locale

**Environment (please complete the following information):**

- OS: linux
- OS version: Linux 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux[uname -a]

**Additional context**
I was trying to set up the development environment to contribute a Pipe for publishing on Sonatype Nexus instances.

Sure thing

```
go test  -failfast -race -coverpkg=./... -covermode=atomic -coverprofile=coverage.txt ./... -run . -timeout=2m
2019/10/10 18:54:42  info loading config file       file=goreleaser.yml
2019/10/10 18:54:42  info RUNNING BEFORE HOOKS     
2019/10/10 18:54:42  info LOADING ENVIRONMENT VARIABLES
2019/10/10 18:54:42  warn pipe skipped              error=publishing is disabled
2019/10/10 18:54:42  info GETTING AND VALIDATING GIT STATE
2019/10/10 18:54:42  info releasing v0.0.2, commit 0d2e386b1f7dcab1065fcc1940927f45cdfe6de7
2019/10/10 18:54:42  warn pipe skipped              error=disabled during snapshot mode
2019/10/10 18:54:42  info PARSING TAG              
2019/10/10 18:54:42  info SETTING DEFAULTS         
2019/10/10 18:54:42  info LOADING ENVIRONMENT VARIABLES
2019/10/10 18:54:42  info SNAPSHOTING              
2019/10/10 18:54:42  info GITHUB/GITLAB/GITEA RELEASES
2019/10/10 18:54:42  info PROJECT NAME             
2019/10/10 18:54:42  info BUILDING BINARIES        
2019/10/10 18:54:42  info ARCHIVES                 
2019/10/10 18:54:42  info LINUX PACKAGES WITH NFPM 
2019/10/10 18:54:42  info SNAPCRAFT PACKAGES       
2019/10/10 18:54:42  info CALCULATING CHECKSUMS    
2019/10/10 18:54:42  info SIGNING ARTIFACTS        
2019/10/10 18:54:42  info DOCKER IMAGES            
2019/10/10 18:54:42  info ARTIFACTORY              
2019/10/10 18:54:42  info S3                       
2019/10/10 18:54:42  info BLOB                     
2019/10/10 18:54:42  info HOMEBREW TAP FORMULA     
2019/10/10 18:54:42  warn optimistically guessing `brew[0].installs`, double check
2019/10/10 18:54:42  info SCOOP MANIFEST           
2019/10/10 18:54:42  info SNAPSHOTING              
2019/10/10 18:54:42  info CHECKING ./DIST          
2019/10/10 18:54:42  info WRITING EFFECTIVE CONFIG FILE
2019/10/10 18:54:42  info writing                   config=dist/config.yaml
2019/10/10 18:54:42  info GENERATING CHANGELOG     
2019/10/10 18:54:42  warn pipe skipped              error=not available for snapshots
2019/10/10 18:54:42  info BUILDING BINARIES        
2019/10/10 18:54:42  info building                  binary=dist/fake_linux_amd64/fake
2019/10/10 18:54:43  info ARCHIVES                 
2019/10/10 18:54:43  info creating                  archive=dist/fake_SNAPSHOT-0d2e386_linux_amd64.tar.gz
2019/10/10 18:54:47  info LINUX PACKAGES WITH NFPM 
2019/10/10 18:54:47  warn pipe skipped              error=no output formats configured
2019/10/10 18:54:47  info SNAPCRAFT PACKAGES       
2019/10/10 18:54:47  warn pipe skipped              error=no summary nor description were provided
2019/10/10 18:54:47  info CALCULATING CHECKSUMS    
2019/10/10 18:54:47  info checksumming              file=fake_SNAPSHOT-0d2e386_linux_amd64.tar.gz
2019/10/10 18:54:47  info SIGNING ARTIFACTS        
2019/10/10 18:54:47  warn pipe skipped              error=artifact signing is disabled
2019/10/10 18:54:47  info DOCKER IMAGES            
2019/10/10 18:54:47  warn pipe skipped              error=docker section is not configured
2019/10/10 18:54:47  info PUBLISHING               
2019/10/10 18:54:47  warn pipe skipped              error=publishing is disabled
--- FAIL: TestReleaseProject (5.20s)
    git.go:14: 
        	Error Trace:	git.go:14
        	            				main_test.go:165
        	            				main_test.go:23
        	Error:      	"Inicializado repositorio Git vacío en /tmp/399617278/.git/
        	            	" does not contain "Initialized empty Git repository"
        	Test:       	TestReleaseProject
FAIL
coverage: 29.8% of statements in ./...
FAIL	github.com/goreleaser/goreleaser	5.450s
ok  	github.com/goreleaser/goreleaser/internal/artifact	1.223s	coverage: 3.3% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/builders/golang	6.710s	coverage: 8.7% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/client	1.367s	coverage: 6.1% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/deprecate	1.205s	coverage: 0.5% of statements in ./...
--- FAIL: TestGit (0.04s)
    git_test.go:18: 
        	Error Trace:	git_test.go:18
        	Error:      	Not equal: 
        	            	expected: "git: 'command-that-dont-exist' is not a git command. See 'git --help'.\n"
        	            	actual  : "git: 'command-that-dont-exist'no es un comando de git. Mira 'git --help'.\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-git: 'command-that-dont-exist' is not a git command. See 'git --help'.
        	            	+git: 'command-that-dont-exist'no es un comando de git. Mira 'git --help'.
        	            	 
        	Test:       	TestGit
FAIL
coverage: 0.7% of statements in ./...
FAIL	github.com/goreleaser/goreleaser/internal/git	0.212s
ok  	github.com/goreleaser/goreleaser/internal/http	2.575s	coverage: 8.3% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/ids	1.191s	coverage: 0.6% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/linux	1.163s	coverage: 0.7% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/middleware	1.261s	coverage: 1.1% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe	1.290s	coverage: 0.4% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/archive	1.889s	coverage: 10.1% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/artifactory	1.441s	coverage: 8.0% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/before	1.957s	coverage: 1.8% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/blob	45.283s	coverage: 4.9% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/brew	1.520s	coverage: 9.2% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/build	1.245s	coverage: 6.9% of statements in ./...
2019/10/10 18:54:57  info loaded custom release notes file=testdata/changes.md
2019/10/10 18:54:57  info loaded custom release notes file=testdata/changes.md
2019/10/10 18:54:57  info writing                   changelog=/tmp/goreleasertest784828270/CHANGELOG.md
--- FAIL: TestChangelog (0.26s)
    git.go:14: 
        	Error Trace:	git.go:14
        	            				changelog_test.go:58
        	Error:      	"Inicializado repositorio Git vacío en /tmp/goreleasertest784828270/.git/
        	            	" does not contain "Initialized empty Git repository"
        	Test:       	TestChangelog
FAIL
coverage: 4.9% of statements in ./...
FAIL	github.com/goreleaser/goreleaser/internal/pipe/changelog	0.429s
ok  	github.com/goreleaser/goreleaser/internal/pipe/checksums	1.163s	coverage: 3.7% of statements in ./...
2019/10/10 18:54:58  info LOADING ENVIRONMENT VARIABLES
2019/10/10 18:54:58  info SNAPSHOTING              
2019/10/10 18:54:58  info GITHUB/GITLAB/GITEA RELEASES
2019/10/10 18:54:58  info PROJECT NAME             
2019/10/10 18:54:58  info BUILDING BINARIES        
2019/10/10 18:54:58  info ARCHIVES                 
2019/10/10 18:54:58  info LINUX PACKAGES WITH NFPM 
2019/10/10 18:54:58  info SNAPCRAFT PACKAGES       
2019/10/10 18:54:58  info CALCULATING CHECKSUMS    
2019/10/10 18:54:58  info SIGNING ARTIFACTS        
2019/10/10 18:54:58  info DOCKER IMAGES            
2019/10/10 18:54:58  info ARTIFACTORY              
2019/10/10 18:54:58  info S3                       
2019/10/10 18:54:58  info BLOB                     
2019/10/10 18:54:58  info HOMEBREW TAP FORMULA     
2019/10/10 18:54:58  warn optimistically guessing `brew[0].installs`, double check
2019/10/10 18:54:58  info SCOOP MANIFEST           
--- FAIL: TestFillBasicData (0.04s)
    git.go:14: 
        	Error Trace:	git.go:14
        	            				defaults_test.go:20
        	Error:      	"Inicializado repositorio Git vacío en /tmp/goreleasertest396673213/.git/
        	            	" does not contain "Initialized empty Git repository"
        	Test:       	TestFillBasicData
FAIL
coverage: 12.4% of statements in ./...
FAIL	github.com/goreleaser/goreleaser/internal/pipe/defaults	0.223s
ok  	github.com/goreleaser/goreleaser/internal/pipe/dist	1.170s	coverage: 1.2% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/docker	39.391s	coverage: 8.4% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/effectiveconfig	1.222s	coverage: 1.2% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/env	1.190s	coverage: 3.3% of statements in ./...
2019/10/10 18:55:04  info releasing v0.0.1, commit eca400c3f6d3b666428b512220943420384f971c
--- FAIL: TestSingleCommit (0.06s)
    git.go:14: 
        	Error Trace:	git.go:14
        	            				git_test.go:31
        	Error:      	"Inicializado repositorio Git vacío en /tmp/goreleasertest667966534/.git/
        	            	" does not contain "Initialized empty Git repository"
        	Test:       	TestSingleCommit
FAIL
coverage: 3.6% of statements in ./...
FAIL	github.com/goreleaser/goreleaser/internal/pipe/git	0.173s
ok  	github.com/goreleaser/goreleaser/internal/pipe/nfpm	1.794s	coverage: 7.8% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/project	1.180s	coverage: 0.8% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/publish	1.166s	coverage: 4.3% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/put	1.356s	coverage: 7.4% of statements in ./...
2019/10/10 18:55:10  info creating or updating release repo=test/test tag=v1.0.0
2019/10/10 18:55:10  info uploading to release      file=/tmp/goreleasertest372680283/bin.deb name=bin.deb
2019/10/10 18:55:10  info uploading to release      file=/tmp/goreleasertest372680283/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:10  info creating or updating release repo=test/test tag=v1.0.0
2019/10/10 18:55:10  info creating or updating release repo=test/test tag=v1.0.0
2019/10/10 18:55:10  warn failed to upload artifact, will retry artifact=bin.tar.gz try=0
2019/10/10 18:55:10  warn failed to upload artifact, will retry artifact=bin.tar.gz try=1
2019/10/10 18:55:10  warn failed to upload artifact, will retry artifact=bin.tar.gz try=2
2019/10/10 18:55:10  warn failed to upload artifact, will retry artifact=bin.tar.gz try=3
2019/10/10 18:55:10  warn failed to upload artifact, will retry artifact=bin.tar.gz try=4
2019/10/10 18:55:10  warn failed to upload artifact, will retry artifact=bin.tar.gz try=5
2019/10/10 18:55:11  warn failed to upload artifact, will retry artifact=bin.tar.gz try=6
2019/10/10 18:55:11  warn failed to upload artifact, will retry artifact=bin.tar.gz try=7
2019/10/10 18:55:11  warn failed to upload artifact, will retry artifact=bin.tar.gz try=8
2019/10/10 18:55:12  warn failed to upload artifact, will retry artifact=bin.tar.gz try=9
2019/10/10 18:55:12  info creating or updating release repo=test/test tag=v1.0.0
2019/10/10 18:55:12  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:12  warn failed to upload artifact, will retry artifact=bin.tar.gz try=0
2019/10/10 18:55:12  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:12  warn failed to upload artifact, will retry artifact=bin.tar.gz try=1
2019/10/10 18:55:12  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:12  warn failed to upload artifact, will retry artifact=bin.tar.gz try=2
2019/10/10 18:55:12  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:12  warn failed to upload artifact, will retry artifact=bin.tar.gz try=3
2019/10/10 18:55:12  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:12  warn failed to upload artifact, will retry artifact=bin.tar.gz try=4
2019/10/10 18:55:13  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:13  warn failed to upload artifact, will retry artifact=bin.tar.gz try=5
2019/10/10 18:55:13  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:13  warn failed to upload artifact, will retry artifact=bin.tar.gz try=6
2019/10/10 18:55:13  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:13  warn failed to upload artifact, will retry artifact=bin.tar.gz try=7
2019/10/10 18:55:14  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:14  warn failed to upload artifact, will retry artifact=bin.tar.gz try=8
2019/10/10 18:55:14  info uploading to release      file=/tmp/goreleasertest859036158/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:14  warn failed to upload artifact, will retry artifact=bin.tar.gz try=9
2019/10/10 18:55:14  info creating or updating release repo=test/test tag=v1.0.0
2019/10/10 18:55:14  info uploading to release      file=/tmp/goreleasertest741681477/bin.tar.gz name=bin.tar.gz
2019/10/10 18:55:14  warn failed to upload artifact, will retry artifact=bin.tar.gz try=0
2019/10/10 18:55:14  info uploading to release      file=/tmp/goreleasertest741681477/bin.tar.gz name=bin.tar.gz
--- FAIL: TestDefault (0.07s)
    git.go:14: 
        	Error Trace:	git.go:14
        	            				release_test.go:170
        	Error:      	"Inicializado repositorio Git vacío en /tmp/goreleasertest348879328/.git/
        	            	" does not contain "Initialized empty Git repository"
        	Test:       	TestDefault
FAIL
coverage: 5.9% of statements in ./...
FAIL	github.com/goreleaser/goreleaser/internal/pipe/release	4.915s
ok  	github.com/goreleaser/goreleaser/internal/pipe/s3	27.593s	coverage: 5.9% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/scoop	1.165s	coverage: 5.3% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/semver	1.148s	coverage: 1.2% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/sign	1.688s	coverage: 4.3% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/snapcraft	18.269s	coverage: 9.0% of statements in ./...
ok  	github.com/goreleaser/goreleaser/internal/pipe/snapshot	1.275s	coverage: 1.6% of statements in ./...
?   	github.com/goreleaser/goreleaser/internal/pipeline	[no test files]
ok  	github.com/goreleaser/goreleaser/internal/semerrgroup	1.424s	coverage: 1.6% of statements in ./...
?   	github.com/goreleaser/goreleaser/internal/static	[no test files]
--- FAIL: TestGit (0.04s)
    git.go:14: 
        	Error Trace:	git.go:14
        	            				git_test.go:10
        	Error:      	"Inicializado repositorio Git vacío en /tmp/goreleasertest935965250/.git/
        	            	" does not contain "Initialized empty Git repository"
        	Test:       	TestGit
FAIL
coverage: 1.8% of statements in ./...
FAIL	github.com/goreleaser/goreleaser/internal/testlib	0.148s
ok  	github.com/goreleaser/goreleaser/internal/tmpl	1.261s	coverage: 2.0% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/archive	1.188s	coverage: 2.8% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/archive/gzip	1.091s	coverage: 1.0% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/archive/targz	1.117s	coverage: 1.2% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/archive/zip	1.106s	coverage: 1.1% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/build	1.078s	coverage: 0.4% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/config	1.085s	coverage: 1.7% of statements in ./...
ok  	github.com/goreleaser/goreleaser/pkg/context	1.068s	coverage: 0.8% of statements in ./...
?   	github.com/goreleaser/goreleaser/pkg/defaults	[no test files]
FAIL
Makefile:20: recipe for target 'test' failed
make: *** [test] Error 1

```

> > We can add `LC_ALL=C` before the git commands, to force the message to be in English.
> 
> maybe directly to makefile even?

```
test:
	LC_ALL=C go test $(TEST_OPTIONS) -failfast -race -coverpkg=./... -covermode=atomic -coverprofile=coverage.txt $(SOURCE_FILES) -run $(TEST_PATTERN) -timeout=2m
```
Solves the issue.
(Created a pull request https://github.com/goreleaser/goreleaser/pull/1195 ) 

Althought now I have another failing test:
```
2019/10/12 11:42:16  warn DEPRECATED: `s3` should not be used anymore, check https://goreleaser.com/deprecations#s3 for more info.
--- FAIL: TestUpload (5.86s)
    s3_test.go:228: waiting for minio to be healthy
    s3_test.go:225: minio is healthy
    s3_test.go:195: creating test bucket
    s3_test.go:202: 
        	Error Trace:	s3_test.go:202
        	            				s3_test.go:99
        	Error:      	Received unexpected error:
        	            	RequestError: send request failed
        	            	caused by: Put http://127.0.0.1:34025/test: http: server closed idle connection
        	Test:       	TestUpload
FAIL
```

The logs of the docker container:
```
$ docker logs frosty_shamir 
Error response from daemon: configured logging driver does not support reading
```

Should I open a new bug report?

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ setup:
 
 # Run all the tests
 test:
-	go test $(TEST_OPTIONS) -failfast -race -coverpkg=./... -covermode=atomic -coverprofile=coverage.txt $(SOURCE_FILES) -run $(TEST_PATTERN) -timeout=2m
+	LC_ALL=C go test $(TEST_OPTIONS) -failfast -race -coverpkg=./... -covermode=atomic -coverprofile=coverage.txt $(SOURCE_FILES) -run $(TEST_PATTERN) -timeout=2m
 .PHONY: test
 
 # Run all the tests and opens the coverage report

